staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -334,38 +347,51 @@ func New(opts Options) (authenticator.Token, error) {
 		}
 	}
 
-	authenticator := &Authenticator{
+	authn := &jwtAuthenticator{
 		jwtAuthenticator: opts.JWTAuthenticator,
-		cancel:           cancel,
 		resolver:         resolver,
 		celMapper:        celMapper,
 		requiredClaims:   requiredClaims,
 	}
+	var noError errorHolder
+	authn.healthCheck.Store(&noError)
```

Comment:
don't we need to be able to distinguish between 1) not-yet-initialized state, 2) error-during-initialized state, and 3) completed-initialization-without-error state?

storing `&noError` here AND on line 384 makes cases 1 and 3 indistinguishable, right?